ssl/statem/statem_clnt.c

```diff
@@ -3336,7 +3336,12 @@ int tls_construct_client_certificate(SSL *s, WPACKET *pkt)
                     SSL3_CC_HANDSHAKE | SSL3_CHANGE_CIPHER_CLIENT_WRITE))) {
         SSLerr(SSL_F_TLS_CONSTRUCT_CLIENT_CERTIFICATE,
                SSL_R_CANNOT_CHANGE_CIPHER);
-        goto err;
+        /*
+         * This is a fatal error, which leaves
+         * enc_write_ctx in an inconsistent state
+         * and thus ssl3_send_alert may crash.
+         */
+        return 0;
```

Comment:
IMHO, it seems something wrong in that alert sending procedure, might we find out and fix the problem instead of just simply returning 0. I dig a little in TLS 1.3 specifications, it says:

> Whenever an implementation encounters a fatal error condition, it SHOULD send an appropriate fatal alert and MUST close the connection without sending or receiving any additional data. 

So it seems an alert should be sent in this situation.